base/utf8proc.jl

```diff
@@ -3,12 +3,12 @@
 # Various Unicode functionality from the utf8proc library
 module UTF8proc
 
-import Base: show, showcompact, ==, hash, string, symbol, isless, length, eltype, start, next, done, convert
+import Base: show, showcompact, ==, hash, string, symbol, isless, length, eltype, start, next, done, convert, isvalid
 
 export isgraphemebreak
 
 # also exported by Base:
-export normalize_string, graphemes, is_valid_char, is_assigned_char, charwidth,
+export normalize_string, graphemes, is_valid_char, is_assigned_char, charwidth, isvalid,
```

Comment:
One thing I like about this commit is that `isvalid` is something that is more general than just strings.  For this reason, does it actually make sense to export it here (from `UTF8proc`) in addition to from `Base`?  The other things exported on this line are string-specific.

Also, in addition to the interfaces you provide, what about being able to call e.g. `isvalid(s::UTF16String)` (without explicitly providing the type as an argument).  This interface would match some current uses of `isvalid` in julia (for instance, in `base/sparse/cholmod.jl`).